cmake/OmimConfig.cmake

```diff
@@ -1,7 +1,6 @@
 # Flags for all
 set(OMIM_WARNING_FLAGS
   $<$<NOT:$<CXX_COMPILER_ID:MSVC>>:-Wall -Wextra -Wpedantic>
-  $<$<NOT:$<CXX_COMPILER_ID:MSVC>>:-Wno-unused-parameter>  # We have a lot of functions with unused parameters
```

Comment:
What are the pros of enabling this warning? I see the following cons:
- Git blame is more complicated.
- Where parameter names are removed, they're not clear and the reader should guess to understand.
- Where parameter names are commented, it is not possible to comment large blocks of code with /* */ anymore.